### ISSUE TITLE
Add an io.Base compatible MediaUpload object.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Dragon-Born/google-api-python-client#123]
```
Add an io.Base compatible MediaUpload object.
```

Original issue reported on code.google.com by `jcgregorio@google.com` on 13 Apr 2012 at 8:41

**Comment by GoogleCodeExporter** [comment]
```
Is there an argument for having a single MediaUpload type that can take any 
io.Base/file-like object?
```

Original comment by `afs...@google.com` on 14 Apr 2012 at 12:20

**Comment by GoogleCodeExporter** [comment]
```
Since we are still targeting Python 2.5 and io.Base only appears in 2.6 and 
later, io.Base support will have to be a feature, not the primary mechanism.
```

Original comment by `jcgregorio@google.com` on 14 Apr 2012 at 12:38

**Comment by GoogleCodeExporter** [comment]
```
Sorry, I meant you could use the intersection of interface between io.Base and 
pre-2.6 file-like objects. They are quite similar.
```

Original comment by `afs...@google.com` on 16 Apr 2012 at 5:01

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I've hacked together slight modifications to HttpRequest and MediaFileUpload 
for sending file objects without specifying their size upfront. Should I clean 
this up and submit a revision to http.py?
```

Original comment by `yova...@google.com` on 1 Jun 2012 at 10:33

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 8ae1f255521a.
```

Original comment by `jcgregorio@google.com` on 12 Jun 2012 at 1:37
- Changed state: **Fixed**
